solidity/contracts/core/pool-diamond/facets/PerpTradeFacet.sol

```diff
@@ -912,7 +943,29 @@ contract PerpTradeFacet is PerpTradeFacetInterface {
       );
     }
 
-    if (!isLong) LibPoolV1.decreaseShortSize(indexToken, position.size);
+    if (!isLong) {
+      if (ds.shortSizeOf[indexToken] == 0)
+        ds.shortAveragePriceOf[indexToken] = vars.markPrice;
+      else {
+        (bool isProfit, uint256 realizedPnl) = GetterFacetInterface(
+          address(this)
+        ).getDeltaWithoutFundingFee(
```

Comment:
on `increasePosition`, reduce collateral use getDelta, is it correct to use w/o fee here?

Reply:
The only fee included in `getDelta` is funding fee which is not desired in this calculation. The formula would only case about the delta from price movement. We already have separated contract states for funding fee accounting.

Reply:
I see.